api/python/quilt3/packages.py

```diff
@@ -1141,24 +1141,30 @@ def _materialize(self, dest_url):
         pkg._meta = self._meta
         # Since all that is modified is physical keys, pkg will have the same top hash
         file_list = []
+        entries = []
         for logical_key, entry in self.walk():
             # Copy the datafiles in the package.
             physical_key = _to_singleton(entry.physical_keys)
+            unversioned_physical_key = physical_key.split('?', 1)[0]
```

Comment:
Object keys may contain `?`
```suggestion
            unversioned_physical_key = physical_key.rsplit('?', 1)[0]
```